lib/_http_outgoing.js

```diff
@@ -426,6 +430,37 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns an array of the names of the current outgoing headers.
+OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
+  return (this._headers ? Object.keys(this._headers) : []);
+};
+
+
+// Returns a shallow copy of the current outgoing headers.
```

Comment:
I think https://github.com/nodejs/node/pull/10795 is a different situation. There is arguably less usefulness in mutating the list of ciphers/hashes supported by OpenSSL. Mutating the array values in this PR is *more* useful because you do not have to continually execute `setHeader()`, which has additional overhead, when you want to add more values for that header.